proposed/2022/cpm-dotnet-cli-add-package-support.md

```diff
@@ -4,218 +4,146 @@
+| 2 | ❌ | ❌ | ❌ | ✔️ | Add `PackageReference` to the project file. Add `PackageVersion` to the `Directory.Packages.Props` file. Use version specified in the commandline. | ✔️ |
+| 3 | ❌ | ❌ | ✔️ | ❌ |  Add `PackageReference` to the project file. No changes to the `Directory.Packages.Props` file. Basically we are reusing the version defined centrally for this package. | ✔️ |
+| 4 | ❌ | ❌ | ✔️ | ✔️ | Add `PackageReference` to the project file. Update `PackageVersion` in the `Directory.Packages.Props` file with the version specified in the commandline.  | ✔️ |
+| 5 | ❌ | ✔️ | ❌ | ❌ | Not a valid scenario because a `VersionOverride` can't exist without `PackageReference`. | ❌ |
+| 6 | ❌ | ✔️ | ❌ | ✔️ | Not a valid scenario because a `VersionOverride` can't exist without `PackageReference`. | ❌ |
+| 7 | ❌ | ✔️ | ✔️ | ❌ | Not a valid scenario because a `VersionOverride` can't exist without `PackageReference`. | ❌ |
+| 8 | ❌ | ✔️ | ✔️ | ✔️ | Not a valid scenario because a `VersionOverride` can't exist without `PackageReference`. | ❌ |
+| 9 | ✔️ | ❌ | ❌ | ❌ | Emit an error -OR- Remove `Version` from `PackageReference`, Add `PackageVersion` to the `Directory.Packages.Props` file. Use `Version` from `PackageReference` if it exists otherwise use latest version from the package sources. | ✔️ |
+| 10 | ✔️ | ❌ | ❌ | ✔️ | Emit an error -OR- Remove `Version` from `PackageReference`, Add `PackageVersion` to the `Directory.Packages.Props` file. Use `Version` passed in the commandline. | ✔️ |
+| 11 | ✔️ | ❌ | ✔️ | ❌ | No-op -OR- Update `PackageVersion` in the `Directory.Packages.Props` file,  use latest version from the package sources. | ✔️ |
+| 12 | ✔️ | ❌ | ✔️ | ✔️ | Update `PackageVersion` in the `Directory.Packages.Props` file, use version specified in the commandline. | ✔️ |
+| 13 | ✔️ | ✔️ |❌  | ❌ | Update `VersionOverride` in the existing `PackageReference` item, use latest version from the package sources. | ✔️ |
+| 14 | ✔️ | ✔️ | ❌ | ✔️ | Update `VersionOverride` in the existing `PackageReference` item, use version specified in the commandline. | ✔️ |
+| 15 | ✔️ | ✔️ | ✔️ | ❌ | Update `VersionOverride` in the existing `PackageReference` item, use latest version from the package sources. | ✔️ |
+| 16 | ✔️ | ✔️ | ✔️ | ✔️ | Update `VersionOverride` in the existing `PackageReference` item, use version specified in the commandline. | ✔️ |
+
+### Technical explanation
+
+NuGet restore operation generates `{projectName}.nuget.dgspec.json` file that file that maintains a project's top-level dependencies along with other metadata for `PackageReference` style projects. 
+- `dgspec.json` file will have `centralPackageVersionsManagementEnabled` property set to `true` for projects onboarded onto CPM.
+- `dotnet add pacakge` command currently access [`ProjectRestoreMetadata`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs) to perform preview restore.
```

Comment:
`dotnet add pacakge` -> `dotnet add package`

Reply:
Thanks Erick. Fixed in the latest commit.